### DISCUSSION TITLE
Server is still not visible

### DISCUSSION THREAD

**Discussion by Dahmos** [lloesche/valheim-server-docker#73] [Q&A]
Hi, first - thx for this wonderful container! Server is running perfectly for more than 8h now and updates/backups are working fine.

But Ive got a problem to make this server visible in the community browser... also tried to set SERVER_PUBLIC 1 to get sure, but still cant find it in Valheim or Steam Server browser. I saved it via favorite server in Steam and its working this way, but it would be wonderful if I could make it visible directly too... any ideas?! 

Thx in advance... 
Dahmos

Oder ist das noch dem EA geschuldet?

**Comment by Toucan-Sam** [comment]
The default setting is public and it definitely works. I'm currently running two containers on the same machine on different ports and both are showing up in the list.

Valheim requires you to open three(3) ports for the game to work. Whatever you set as SERVER_PORT (eg, 2456), you need to make sure you open your firewall or port forward the next two(2) sequential ports. So, 2456, 2457, and 2458 in total (aka 2456-2458) via UDP.

Also, the in game server browser needs some work. You may have to choose 'Community' and refresh the server list a couple of times for all the servers to come up.

**Comment by Dahmos** [comment]
Yeah, I know default is 1. It was only a try to see if something changes. But I'm completly new to this.

Ive set the 2456-2458 UDP in the docker contqiner as shown in the screens. The server runs on my NAS and I forwarded the 2456-2458 UDP ports in my router to the NAS and allowed it to port forward by itself. I also allowed the firewall on my NAS any connection via TCP and UDP as well...  did I forgot something?

! I'm a bit confused about the firewall on my NAS. Im only able to allow all connections and cant specify ports. Its a Terramaster f2 220. Any suggestions about this?

**Comment by Toucan-Sam** [comment]
It sounds like you've done all the right things. I've never used a Terramaster NAS before so I can't offer any advice there.

If one of your friends is able to join it from the public internet, then I'd say it's probably in the list and you scrolled past it or the in-game server browser is being dumb. :)

**Comment by lloesche** [comment]
Same thing here. Used to be that my server would show up when I searched for the name but I haven't seen my server's name in the public list for a week or so. Ever since the game got so popular and more people started playing it's not been listed. I also noticed that the list of public servers stayed pretty constant around 5k while there's 500k people playing the game at any given point in time right now.

**Comment by Dahmos** [comment]
So you've got a static IP I guess? My Problem is that my IP always changes and so I have to inform my mates in our Whatsapp group about the actual IP if they want to join my server...

**Reply by lloesche** [reply]
I do, and I never considered the use case you're describing where the IP changes. I created https://github.com/lloesche/valheim-server-docker/issues/74 to implement dynamic dns support into the container image. This will give your friends a static hostname they can add to the Steam server favorites.

**Reply by Dahmos** [reply]
This would be great! Thanks a lot in advance!

But Im also thinking about getting a static IP... our provider here offers that for about 5 EUR per month.

**Reply by lloesche** [reply]
After looking into the complexity of solving this properly I figured it makes more sense to refer users with dynamic IP to existing Docker images dedicated to updating dynamic dns. I found this repository that seems to support a wide variety of ddns providers and is also well maintained. Last update was 11h ago: https://github.com/qdm12/ddns-updater

**Comment by Dahmos** [comment]
Ive forwarded the Steam Ports in my Router to my NAS... now it works fine. The NAS doesnt do it by itself while its not supposed to be a Server.
